pypsa/linopf.py

```diff
@@ -1038,5 +1039,7 @@ def save_optimal_capacities(n, iteration, status):
     n.links[['p_nom', 'p_nom_extendable']] = n.links['p_nom_opt'], False
     kwargs['warmstart'] = False
     network_lopf(n, snapshots, **kwargs)
+    if track_iterations:
+        save_optimal_capacities(n, iteration, status)
```

Comment:
```suggestion
```
This seems to be a duplicate of line 1030/1031

Reply:
it is, but otherwise, after the last lopf, the duals remain un-iterated as network_lopf is run again in line 1041. (lines 1026/27 within the iterative optimisation process before), so I prefer to rename the mus again for consistency, but we can leave the last iteration un-indexed as well

Reply:
Alright, I get you, but on the other hand the 'last' index is exactly the final outcome (resulting mu), right?

Reply:
It isn't for the global_constraints (there is no mu without index), but it is for the other results that are being tracked (such as objective, status, etc.) I think it's easier to remove these two lines, you're right